### PR TITLE
[WIP] minimal constant folding

### LINKED ISSUE #6805: Add `--inlineConsts` flag
New flag `--inlineConsts`

Input:

``` ts
const x = 'hello';

console.log(x);
```

Output:

``` ts
console.log('hello');
```

See also #6678 + #6804

TBD: It's not clear you would want longer strings to inline into their use sites. Discuss.

### PR DESCRIPTION
**Pull request by Kingwl**
Fixes #6805

### CODE REVIEW

**Comment by Kingwl** [comment]
ping @RyanCavanaugh @DanielRosenwasser @andy-ms @mhegazy 
Sorry for disturbing you 
i'm going to work on this if this issue accept pr now

File: src/compiler/types.ts [hunk 1]
```diff
@@ -839,6 +839,8 @@ namespace ts {
         exclamationToken?: ExclamationToken;  // Optional definite assignment assertion
         type?: TypeNode;                      // Optional type annotation
         initializer?: Expression;             // Optional initializer
+        resolvedValue?: Expression;
```
[the remark this answers is not in the document]
Comment (Kingwl) [reply src/compiler/types.ts]: This is not a final version, just an initialization 😢

**Comment by mhegazy** [comment]
// cc @rbuckton

**Comment by Kingwl** [comment]
🤷🏻‍♂️ seems not accept pr

**Comment by weswigham** [comment]
I think that comment counts as wanting to continue working on it.

Still cc @rbuckton  I suppose.

**Comment by weswigham** [comment]
@Kingwl This does need a rebase off current master, though. We recently developed a strip whitespace writer, and @rbuckton has been looking into type-based property minification - so by their powers combined you should be capable of some nice things.

**Comment by weswigham** [comment]
@Kingwl Also, since this is still marked as `[WIP]` (and `minimal`), you should probably post about what parts you find concerning and what you want feedback on.

**Comment by RyanCavanaugh** [comment]
Going to close this for the time being - I want to take the original bug back for discussion to figure out why we (I?) thought we needed it.

**Pull request closed without merging**